### PR TITLE
Include log_level as key in template

### LINKED ISSUE #658: Propagate log-level down through workflow.
Gordo reads the log-level from an environment variable (`LOG_LEVEL`), defaulting to debug. We should be able to set this from the config file, in the `runtime.model-builder / model-server etc` sections. It should be possible to set it in the `globals` part for all, or in each specific model (for the builder / client), and then the workflow needs to set for the corresponding component. 

Also, I propose we change the variable from `LOG_LEVEL` to e.g. `GORDO_LOG_LEVEL`.
Feel free to split this into several tasks, e.g. for the builder first.

### PR DESCRIPTION
**Pull request by EinarJohnsen**
Let the user specify the log_level as a key in globals to override the default info log-level in the __init__ of gordo-components. Closes #658

### CODE REVIEW

File: gordo_components/cli/workflow_generator.py
Comment (epa095) [inline gordo_components/cli/workflow_generator.py]: This will overwrite the log-level provided in the CLI

Comment (epa095) [reply gordo_components/cli/workflow_generator.py]: Also, remember to set the loglevel of gordo-deploy itself to the provided loglevel...

**Review by epa095 — COMMENTED** [review]
:-)

**Comment by epa095** [comment]
Whats the status here  @EinarJohnsen ? Do you want some encouragement ? :tada:  :taco:  :1st_place_medal:

**Comment by epa095** [comment]
Note that this is effected by https://github.com/equinor/gordo-components/issues/773

**Comment by EinarJohnsen** [comment]
Thinking about how it should work with the new changes from @milesgranger :thinking:. Should each part of the project use it's own basis-logger? @epa095

File: run_workflow_and_argo.sh
Comment (epa095) [inline run_workflow_and_argo.sh]: Nja, I dont think this can now be overridden by an environment variable anymore

File: gordo_components/cli/workflow_generator.py
Comment (epa095) [inline gordo_components/cli/workflow_generator.py]: Now it is a bit confusing, since there is both `LOG_LEVEL` and `{PREFIX}_LOG_LEVEL`, one influencing the running instance and the other only setting the variable for generated workflow. Also, the workflow has the environment variable as `GORDO_LOG_LEVEL`, but that variable is no longer read:-p

File: run_workflow_and_argo.sh
Comment (EinarJohnsen) [reply run_workflow_and_argo.sh]: I removed this, did not make sense after thinking about it.

File: gordo_components/cli/workflow_generator.py
Comment (epa095) [inline gordo_components/cli/workflow_generator.py]: why not the one from the context? With this one can set the loglevel using the CLI  ``--log-level`` but then get, or not get, it set on the workflow depending on what the environment variable is.

Comment (EinarJohnsen) [reply gordo_components/cli/workflow_generator.py]: I guess it should be possible, but I am not sure how to do it correctly with click. It is not part of the ctx in the workflow, and I have not been able to pass/retrieve it from the CLI context. Perhaps @milesgranger knows something about this.

Comment (ryanjdillon) [reply gordo_components/cli/workflow_generator.py]: You could possibly store a [local config file,](https://click.palletsprojects.com/en/7.x/utils/#finding-application-folders) which could then be read by the sub-CLIs.  Maybe wrap things up in a decorator and ensure is transient.

Comment (epa095) [reply gordo_components/cli/workflow_generator.py]: You have tried to use `@click.pass_context`?

Comment (EinarJohnsen) [reply gordo_components/cli/workflow_generator.py]: @epa095 I have talked to our American friend @milesgranger about it, and he also suggested passing the context. I tried it before, but probably did not pass it far enough down the stream to be able to retrieve it in the workflow_generator. 
@ryanjdillon Good suggestion, but I think if we can use click we should, it's probably just me doing it wrong :man_shrugging:

File: gordo_components/cli/cli.py
Comment (milesgranger) [inline gordo_components/cli/cli.py]: I think at this level, to include each subcommand(group) you'd pass `@click.pass_context` or does it already magically work? I see the tests pass, are we waiting for anything else?

**Comment by flikka** [comment]
@EinarJohnsen what's the status on this?

**Comment by EinarJohnsen** [comment]
It's currently not working as I want it to. I will look more into it after fixing the Python version PR.

File: gordo_components/cli/workflow_generator.py
Comment (milesgranger) [inline gordo_components/cli/workflow_generator.py]: As much as I love a good comment.

File: tests/gordo_components/workflow/test_workflow_generator/test_workflow_generator.py
Comment (milesgranger) [inline tests/gordo_components/workflow/test_workflow_generator/test_workflow_generator.py]: Forget to remove?

**Review by milesgranger — COMMENTED** [review]
Looking good, just a couple comments.

File: gordo_components/cli/workflow_generator.py
Comment (EinarJohnsen) [reply gordo_components/cli/workflow_generator.py]: :upside_down_face:

File: tests/gordo_components/workflow/test_workflow_generator/test_workflow_generator.py
Comment (EinarJohnsen) [reply tests/gordo_components/workflow/test_workflow_generator/test_workflow_generator.py]: Correct, sir! :tophat:

File: gordo_components/cli/cli.py
Comment (ryanjdillon) [inline gordo_components/cli/cli.py:67]: Oh matplotlib... you devil. :smile:  Should this possibly be set to the global runtime level also?

Comment (EinarJohnsen) [reply gordo_components/cli/cli.py:67]: The "problem" by doing that is if we use the log_key and set it to debug, matplotlib will nag us with a wall of text. I think me and @epa095 talked about this and agreed to set it to INFO for now, otherwise it kind of destroys what we did in https://github.com/equinor/gordo-components/issues/657.
I think having another key for this is kind of unnecessary. I think we rather should turn it on manually if it ever gets a problem? But I am open to suggestions. @milesgranger @ryanjdillon @flikka

File: gordo_components/cli/workflow_generator.py
Comment (milesgranger) [inline gordo_components/cli/workflow_generator.py]: Nice change here. :+1:

File: tests/gordo_components/cli/test_cli.py
Comment (milesgranger) [inline tests/gordo_components/cli/test_cli.py]: _very_ minor, but you could have just done `"TEST_LOG_LEVEL" in value`

File: tests/gordo_components/workflow/test_workflow_generator/test_workflow_generator.py
Comment (milesgranger) [inline tests/gordo_components/workflow/test_workflow_generator/test_workflow_generator.py]: Same thing, `log_level in value`

**Review by milesgranger — APPROVED** [review]
Looks goodie to me, just a few comments, you're free to update and merge.

File: tests/gordo_components/cli/test_cli.py
Comment (EinarJohnsen) [reply tests/gordo_components/cli/test_cli.py]: `True - True` Will fix it

**Comment by EinarJohnsen** [comment]
@milesgranger Did a small update, as I saw the input was never actually checked for being uppercase. I guess it's ok, but if you wanna check over feel free. I can wait until the morning to squash and merge.

**Comment by milesgranger** [comment]
Good eye! Yep, it continues to seem good to me. :+1:

**Pull request merged**
